zcash_extensions/src/consensus/transparent.rs

```diff
@@ -0,0 +1,122 @@
+//! Consensus logic for Transparent Zcash Extensions.
+
+use std::convert::TryFrom;
+use zcash_primitives::consensus::NetworkUpgrade;
+use zcash_primitives::extensions::transparent::{Error, Extension, Precondition, Witness};
+use zcash_primitives::transaction::{components::TzeOut, Transaction};
+
+use crate::transparent::demo;
+
+/// Wire value for the demo extension identifier.
+pub const EXTENSION_DEMO: u32 = 0;
+
+/// The set of programs that have assigned type IDs within the Zcash consensus rules.
+#[derive(Debug, Clone, Copy)]
+pub enum ExtensionId {
+    Demo,
+}
+
+pub struct InvalidExtId(u32);
+
+impl TryFrom<u32> for ExtensionId {
+    type Error = InvalidExtId;
+
+    fn try_from(t: u32) -> Result<Self, Self::Error> {
+        match t {
+            EXTENSION_DEMO => Ok(ExtensionId::Demo),
+            n => Err(InvalidExtId(n)),
+        }
+    }
+}
+
+impl From<ExtensionId> for u32 {
+    fn from(type_id: ExtensionId) -> u32 {
+        match type_id {
+            ExtensionId::Demo => EXTENSION_DEMO,
+        }
+    }
+}
+
+/// The complete set of context data that is available to any extension having
+/// an assigned extension type ID. This type may be modified in the future if
+/// additional context information is required by newly integrated TZEs.
+pub struct Context<'a> {
```

Comment:
Actually, I don't think we need to expose `Context` at all, now that you mention it. Extensions don't know about it, and this is really just the shim for inclusion in zcashd - it might even be more appropriate to move this module to zcash/zcash.